app/src/debug/java/com/sample/neuroid/us/activities/MainActivity.kt

```diff
@@ -41,5 +46,37 @@ class MainActivity : AppCompatActivity() {
                 NeuroID.getInstance()?.closeSession()
             }
         }
+        if (!isLocationPermissionGiven()) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_REQUEST_CODE)
+        } else {
+            // since permission is given after SDK start, we need to
+            // resync the location
+            NeuroID.getInstance()?.syncLocation()
+        }
+    }
```

Comment:
is this something we have to put in out docs? or could is there a way we could have it recalled on each metadata event to check?

Reply:
yeah, we can indeed call getLocation() in NIDMetaData.toJson() to get the latest location when a metadata event is generated. This would remove the need for a location listener to get the current location. Adding this and testing.

Reply:
I didn't see this in the latest changes, did I miss it?

Reply:
you missed it. line 146 in NIDMetaData.kt